src/pages/studyView/StudyViewPageStore.ts

```diff
@@ -6405,6 +6405,65 @@ export class StudyViewPageStore
         );
     }
 
+    @computed get chartMetaSetForClinicalAttributes(): {
+        [id: string]: ChartMeta;
+    } {
+        let _chartMetaSet = _.fromPairs(this._customCharts.toJSON());
+        if (_.isEmpty(this.molecularProfiles.result)) {
+            delete _chartMetaSet[
+                SpecialChartsUniqueKeyEnum.GENOMIC_PROFILES_SAMPLE_COUNT
+            ];
+        }
+        // Add meta information for clinical attributes (includes survival attributes)
+        // Convert to a Set for easy access and to update attribute meta information (would be useful while adding new features)
+        _.reduce(
```

Comment:
try to always keep to a functional paradigm where _.someOperation only ever assigns to a new variable.  we then merge those at the end.    i.e.:

```
const clinicalWhatevers = _.reduce(...);
const survivalWhatevers = _.reduce(...);

return { ...clinicalWhatevers, ...surivalWhatevers };

```

Reply:
another way to think about this is in terms of immutability.  We want to always err on the side of operating on/returning a NEW collection of things rather than mutating something in place.   You are effectively doing that here by mutating ```let _chartMetaSet = _.fromPairs(this._customCharts.toJSON());```.    It so happens that this is effectively a new thing because we're calling toJSON and fromPairs etc.  But it would be more clear if we were doing the following:
```
const customCarts = ...
const clinicalWhatevers = _.reduce()
const suruvivaWhaters = _.reduce()

return { ... }; /// this is the immutable here.  It's a NEW object and we run no risk of mistakenly updating state.  

```